epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/madgraph/iolibs/template_files/gpu/MatrixElementKernels.cc

```diff
@@ -23,6 +24,100 @@ namespace mg5amcCpu
 {
   //--------------------------------------------------------------------------
 
+  MatrixElementKernelBase::MatrixElementKernelBase( const BufferMomenta& momenta,         // input: momenta
```

Comment:
Hi @oliviermattelaer you mean why we need MatrixElementKernelBase at all?

This was described in my presentation January 2022, https://indico.cern.ch/event/1103955/

![image](https://github.com/user-attachments/assets/01676bbf-eab6-4e1a-8466-6da6c1d976bb)
![image](https://github.com/user-attachments/assets/e4aaa091-0a49-4654-9972-c5e8f8b63487)
![image](https://github.com/user-attachments/assets/79d3901e-6114-4456-a2e5-9cdc7194491b)

I hope this answers the question? Marking as resolved, feel free to reopen otherwise.